Sources/Sentry/SentryTracer.m

```diff
@@ -63,6 +63,7 @@ @implementation SentryTracer {
     NSMutableDictionary<NSString *, SentryMeasurementValue *> *_measurements;
     dispatch_block_t _idleTimeoutBlock;
     NSMutableArray<id<SentrySpan>> *_children;
+    SentryTransaction *_transaction;
```

Comment:
`h`: I think this introduces a strong reference cycle between transaction and tracer.

Reply:
changing this to `__weak` changes some behavior, need to look more into what is going on with that 😵‍💫

Reply:
I have a proposal for breaking this cycle. I think the profiler doesn't need a reference to the transaction. The profiler needs 

- [`startTimestamp`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryProfiler.mm#L591-L594) - the tracer gives you that info
- [`timestamp`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryProfiler.mm#L597-L600) - the tracer gives you that info
- [`platform`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryProfiler.mm#L536)- is always [cocoa](https://github.com/getsentry/sentry-cocoa/blob/062f06c9d3d15800a2cc1471a6605e2e9dda2730/Sources/Sentry/SentryEvent.m#L40). We could define a constant for this, and use it both in `SentryEvent` and `SentryProfiler`
- [`trace.context.traceId.sentryIdString`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryProfiler.mm#L603) - the tracer gives you that info
- [`transaction`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryProfiler.mm#L604) - is [`tracer.transactionContext.name`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryTracer.m#L577)
- [`eventId.sentryIdString`](https://github.com/getsentry/sentry-cocoa/blob/386b191f24fbc43d1cfc1ae4476b6fbae9953562/Sources/Sentry/SentryProfiler.mm#L602) - this is the only info the tracer doesn't have

The only reason why we need the transaction in the tracer is the `eventId`, as you already pointed out in a Slack discussion some time ago. Instead of keeping creating the `SentryTransaction` earlier and keeping a reference to it, we could do the following:

1. create the `eventId` of the transaction early on and keep a reference to it here.
2. The `SentryProfiler` keeps a dictionary of `eventId` to `SentryTracer`.
3. `SentryProfiler linkTransaction`, `SentryProfiler dropTransaction`, etc. would update the previously mentioned dictionary.
4. We need to create an extra initializer for `SentryTransaction:initWithTrace:eventId`, which calls also a new project internal initializer on the `SentryEvent:initWithEventId`.

How does that sound?

Reply:
Yes, this sounds good. Working on this now.

Reply:
there's also the thread that a transaction starts on, i'll see if i can also track that the same way as this

Reply:
So actually because there's already a `SentryTransactionContext` passed into the `SentryTracer` init, we can just use that. We do need the actual transactions at the end of the profile thought because we're sending up other information about it in the profiling payload, like start/stop timestamps.

But also, I noticed that the only reason we even keep a reference from `SentryTransaction`->`SentryTracer` is so that `SentryTransaction.serialize` can call `SentryTracer.serialize`. But, we could also just provide the resulting dictionary from `SentryTracer.serialize` to the transaction when we initialize it from the tracer. Then we don't need that back reference at all.

Reply:
OK I see that there are a few other things passed back from the tracer to the transaction, but I really don't get why these all need to be separated between these two different classes if they're so intertwined anyways. I also see that `SentryTracer.data` seems to be included into the serialization twice; once in the `contexts.trace` field and again, although this time sanitized, under `extra`. Why the near-duplication?

Reply:
OK, removed the need to keep the reference to the transaction. See https://github.com/getsentry/sentry-cocoa/commit/211a5d1925d269749faef441878226b727eb229a